flink-runtime/src/main/java/org/apache/flink/runtime/scheduler/SchedulerBase.java

```diff
@@ -746,9 +741,7 @@ private boolean isNotifiable(
         return false;
     }
 
-    protected void updateTaskExecutionStateInternal(
-            final ExecutionVertexID executionVertexId,
-            final TaskExecutionStateTransition taskExecutionState) {}
+    protected abstract void onTaskExecutionStateUpdate(final Execution execution);
```

Comment:
How about move the implementation here and provide two abstract method `onTaskFinished` and `onTaskFailed` ？

Reply:
I will rework it along with `isNotifiable`.